pkg/controller/iam/user/controller.go

```diff
@@ -258,3 +254,65 @@ func (t *tagger) Initialize(ctx context.Context, mgd resource.Managed) error {
 	}
 	return errors.Wrap(t.kube.Update(ctx, cr), errKubeUpdateFailed)
 }
+
+func (e *external) updateUser(ctx context.Context, observed *awsiam.GetUserOutput, cr *v1beta1.User) error {
+	if aws.ToString(observed.User.Path) != aws.ToString(cr.Spec.ForProvider.Path) {
+		_, err := e.client.UpdateUser(ctx, &awsiam.UpdateUserInput{
+			NewPath:  cr.Spec.ForProvider.Path,
+			UserName: aws.String(meta.GetExternalName(cr)),
+		})
+
+		return awsclient.Wrap(err, errUpdate)
+	}
+
+	return nil
+}
+
+func (e *external) updatePermissionsBoundary(ctx context.Context, observed *awsiam.GetUserOutput, cr *v1beta1.User) error {
+	boundaryArn := ""
+	var err error
+
+	if observed.User.PermissionsBoundary != nil {
+		boundaryArn = *observed.User.PermissionsBoundary.PermissionsBoundaryArn
+	}
+	if aws.ToString(&boundaryArn) != aws.ToString(cr.Spec.ForProvider.PermissionsBoundary) {
+		if aws.ToString(cr.Spec.ForProvider.PermissionsBoundary) == "" {
+			_, err = e.client.DeleteUserPermissionsBoundary(ctx, &awsiam.DeleteUserPermissionsBoundaryInput{
+				UserName: aws.String(meta.GetExternalName(cr)),
+			})
+		} else {
+			_, err = e.client.PutUserPermissionsBoundary(ctx, &awsiam.PutUserPermissionsBoundaryInput{
+				PermissionsBoundary: cr.Spec.ForProvider.PermissionsBoundary,
+				UserName:            aws.String(meta.GetExternalName(cr)),
+			})
+		}
+
+		return awsclient.Wrap(err, errUpdate)
```

Comment:
To better identify the cause of the error `errUpdate` should be replaced with a more specific error message like `errPutUserPermissionsBoundary` and `errDeleteUserPermissionsBoundary.